### ISSUE TITLE
400 Error creating certificates

### ISSUE THREAD

**Issue by pgjones** [terraform-providers/terraform-provider-acme#71]
I am unable to create a certificate using this provider, I have in the past with the same terraform setup, but the latest version fails for me with `acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for example.com`. (Note I didn't actually try to do this for example.com, but rather for a domain I control and can change the DNS settings for).

I've reached the limit of my knowledge on this, and I think it is an issue with this provider or lego. Hopefully I've provided enough details below that you'll be able to figure it out,
```
Terraform v0.11.13
+ provider.acme v1.3.4
+ provider.gandi (unversioned)
+ provider.gitlab v2.2.0
+ provider.heroku v2.0.1
+ provider.tls v2.0.1
```

```
2019-06-26T09:38:22.452+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:22 [DEBUG] lego: acme: Trying to resolve account by key
2019-06-26T09:38:23.069+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:23 [DEBUG] lego: [example.com, www.example.com] acme: Obtaining bundled SAN certificate
2019-06-26T09:38:24.091+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/OmXGbHjG7C0VGC1fNw9Pued8kUdRewatmB_rHYq1g24
2019-06-26T09:38:24.091+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [www.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/5_qjEloHKj7SYYcb1LgzjAWOckZu0IIArrZbm5SkJQU
2019-06-26T09:38:24.091+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [example.com] acme: Could not find solver for: tls-alpn-01
2019-06-26T09:38:24.092+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [example.com] acme: Could not find solver for: http-01
2019-06-26T09:38:24.092+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [example.com] acme: use dns-01 solver
2019-06-26T09:38:24.092+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [www.example.com] acme: Could not find solver for: tls-alpn-01
2019-06-26T09:38:24.092+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [www.example.com] acme: Could not find solver for: http-01
2019-06-26T09:38:24.092+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [www.example.com] acme: use dns-01 solver
2019-06-26T09:38:24.092+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:24 [DEBUG] lego: [example.com] acme: Preparing to solve DNS-01
2019-06-26T09:38:25.321+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:25 [DEBUG] lego: API response: DNS Record Created
2019-06-26T09:38:25.321+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:25 [DEBUG] lego: [www.example.com] acme: Preparing to solve DNS-01
2019-06-26T09:38:26.140+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:26 [DEBUG] lego: API response: DNS Record Created
2019-06-26T09:38:26.140+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:26 [DEBUG] lego: [example.com] acme: Trying to solve DNS-01
2019-06-26T09:38:26.140+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:26 [DEBUG] lego: [example.com] acme: Checking DNS record propagation using [8.8.8.8:53 8.8.4.4:53]
2019-06-26T09:38:26.140+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:26 [DEBUG] lego: Wait for propagation [timeout: 20m0s, interval: 20s]
2019-06-26T09:38:27.117+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:27 [DEBUG] lego: [www.example.com] acme: Trying to solve DNS-01
2019-06-26T09:38:27.117+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:27 [DEBUG] lego: [www.example.com] acme: Checking DNS record propagation using [8.8.8.8:53 8.8.4.4:53]
2019-06-26T09:38:27.117+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:27 [DEBUG] lego: Wait for propagation [timeout: 20m0s, interval: 20s]
2019/06/26 09:38:27 [TRACE] dag/walk: vertex "meta.count-boundary (count boundary fixup)", waiting for: "acme_certificate.ocean"
2019/06/26 09:38:27 [TRACE] dag/walk: vertex "root", waiting for: "meta.count-boundary (count boundary fixup)"
2019/06/26 09:38:27 [TRACE] dag/walk: vertex "provider.acme (close)", waiting for: "acme_certificate.ocean"
2019-06-26T09:38:28.083+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:28 [DEBUG] lego: [example.com] acme: Cleaning DNS-01 challenge
2019-06-26T09:38:28.391+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:28 [DEBUG] lego: [www.example.com] acme: Cleaning DNS-01 challenge
2019-06-26T09:38:28.903+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:28 [DEBUG] lego: Unable to deactivated authorizations: https://acme-v02.api.letsencrypt.org/acme/authz/OmXGbHjG7C0VGC1fNw9Pued8kUdRewatmB_rHYq1g24
2019-06-26T09:38:29.142+0100 [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 2019/06/26 09:38:29 [DEBUG] lego: Unable to deactivated authorizations: https://acme-v02.api.letsencrypt.org/acme/authz/5_qjEloHKj7SYYcb1LgzjAWOckZu0IIArrZbm5SkJQU
2019/06/26 09:38:29 [TRACE] root: eval: *terraform.EvalWriteState
2019/06/26 09:38:29 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2019/06/26 09:38:29 [TRACE] root: eval: *terraform.EvalIf
2019/06/26 09:38:29 [TRACE] root: eval: *terraform.EvalWriteState
2019/06/26 09:38:29 [TRACE] root: eval: *terraform.EvalWriteDiff
2019/06/26 09:38:29 [TRACE] root: eval: *terraform.EvalApplyPost
2019/06/26 09:38:29 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* acme_certificate.ocean: error creating certificate: acme: Error -> One or more domains had a problem:
[example.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.example.com, url: 
[www.example.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.www.example.com, url: 
2019/06/26 09:38:29 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* acme_certificate.ocean: error creating certificate: acme: Error -> One or more domains had a problem:
[example.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.example.com, url: 
[www.example.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.www.example.com, url: 
2019/06/26 09:38:29 [TRACE] [walkApply] Exiting eval tree: acme_certificate.ocean
2019/06/26 09:38:29 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2019/06/26 09:38:29 [TRACE] dag/walk: upstream errored, not walking "provider.acme (close)"
2019/06/26 09:38:29 [TRACE] dag/walk: upstream errored, not walking "root"
2019/06/26 09:38:29 [TRACE] Preserving existing state lineage "55ee690f-7b13-e2c8-97bb-a6735508293e"
2019/06/26 09:38:29 [TRACE] Preserving existing state lineage "55ee690f-7b13-e2c8-97bb-a6735508293e"
2019/06/26 09:38:29 [TRACE] Preserving existing state lineage "55ee690f-7b13-e2c8-97bb-a6735508293e"
2019/06/26 09:38:29 [TRACE] Preserving existing state lineage "55ee690f-7b13-e2c8-97bb-a6735508293e"

2019/06/26 09:38:29 [DEBUG] plugin: waiting for all plugin processes to complete...
Error: Error applying plan:

1 error(s) occurred:

* acme_certificate.ocean: 1 error(s) occurred:

* acme_certificate.ocean: error creating certificate: acme: Error -> One or more domains had a problem:
[example.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.example.com, url: 
[www.example.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.www.example.com, url: 
```

**Comment by vancluever** [comment]
@pgjones sounds like you are running into Google DNS provider problems that have been fixed in the latest lego release. We will be releasing a new version this this in soon, so keep an eye out!

**Comment by vancluever** [comment]
Should be fixed with the v1.4.0 release.
